core/templates/components/forms/custom-forms-directives/thumbnail-uploader.component.spec.ts

```diff
@@ -0,0 +1,365 @@
+  let imageLocalStorageService: ImageLocalStorageService;
+  let alertsService: AlertsService;
+  let ngbModal: NgbModal;
+
+  beforeEach(waitForAsync(() => {
+    TestBed.configureTestingModule({
+      imports: [HttpClientTestingModule],
+      declarations: [
+        ThumbnailUploaderComponent,
+        EditThumbnailModalComponent,
+        MockTranslatePipe
+      ],
+      providers: [ImageUploadHelperService],
+      schemas: [NO_ERRORS_SCHEMA]
+    }).compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(ThumbnailUploaderComponent);
+    component = fixture.componentInstance;
+    contextService = TestBed.inject(ContextService);
+    imageUploadHelperService = TestBed.inject(ImageUploadHelperService);
+    assetsBackendApiService = TestBed.inject(AssetsBackendApiService);
+    imageLocalStorageService = TestBed.inject(ImageLocalStorageService);
+    alertsService = TestBed.inject(AlertsService);
+    ngbModal = TestBed.inject(NgbModal);
+
+    spyOn(contextService, 'getEntityType').and.returnValue('exploration');
+    spyOn(contextService, 'getEntityId').and.returnValue('expId');
+  });
+
+  it('should set uploaded and editable thumbnail on initialization', () => {
+    component.filename = 'thumbnail-1';
+
+    expect(component.editableThumbnailDataUrl).toBe(undefined);
+    expect(component.uploadedImage).toBe(undefined);
+
+    component.ngOnInit();
+
+    expect(component.editableThumbnailDataUrl)
+      .toBe('/assetsdevhandler/exploration/expId/assets/thumbnail/thumbnail-1');
+    expect(component.uploadedImage)
+      .toBe('/assetsdevhandler/exploration/expId/assets/thumbnail/thumbnail-1');
+  });
+
+  it('should update the thumbnail image when thumbnail filename' +
+    ' changes', () => {
+    component.filename = 'thumbnail-1';
+    let changes: SimpleChanges = {
+      filename: {
+        currentValue: 'thumbnail-2',
+        previousValue: 'thumbnail-1',
+        firstChange: false,
+        isFirstChange: () => false
+      }
+    };
+    component.ngOnInit();
+
+    expect(component.editableThumbnailDataUrl)
+      .toBe('/assetsdevhandler/exploration/expId/assets/thumbnail/thumbnail-1');
+
+    component.ngOnChanges(changes);
+
+    expect(component.editableThumbnailDataUrl)
+      .toBe('/assetsdevhandler/exploration/expId/assets/thumbnail/thumbnail-2');
+  });
+
+  it('should not update the thumbnail image when new thumbnail is same as' +
+    ' the old one', () => {
+    spyOn(
+      imageUploadHelperService, 'getTrustedResourceUrlForThumbnailFilename');
+    component.filename = 'thumbnail-1';
+    let changes: SimpleChanges = {
+      filename: {
+        currentValue: 'thumbnail-1',
+        previousValue: 'thumbnail-1',
+        firstChange: true,
+        isFirstChange: () => true
+      }
+    };
+
+    component.ngOnChanges(changes);
+
+    expect(imageUploadHelperService.getTrustedResourceUrlForThumbnailFilename)
+      .not.toHaveBeenCalled();
```

Comment:
Should we check that the "editableThumbnailDataUrl" remains the same?

Reply:
Done